### PR TITLE
ECAL - Fix ECAL static analyzer issues in CalibCalorimetry and CondFormats packages

### PR DESCRIPTION
**Pull request by thomreis**
#### PR description:

Fixes some ECAL related static analyzer issues in CalibCalorimetry and CondFormats packages

- CalibCalorimetry/CaloMiscalibTools
- CalibCalorimetry/EcalCorrelatedNoiseAnalysisAlgos
- CalibCalorimetry/EcalLaserSorting
- CondFormats/EcalCorrections

#### PR validation:

Passes matrix tests.

### CODE REVIEW

**Comment by tvami** [comment]
Hi @thomreis can you confirm my understanding that none of this is needed for the data taking? Thanks! Please mention in the PR description that there is no backport needed.

**Comment by thomreis** [comment]
None of this is needed for data taking. No backport is needed.

**Comment by tvami** [comment]
Thanks for confirming!

File: CalibCalorimetry/CaloMiscalibTools/src/MiscalibReaderXMLEcalEndcap.cc [hunk 1]
```diff
@@ -9,18 +9,10 @@ DetId MiscalibReaderFromXMLEcalEndcap::parseCellEntry(XERCES_CPP_NAMESPACE::DOMN
...
-  }
-
-  catch (...)
-
-  {
+  if (EEDetId::validDetId(ix, iy, iz)) {
+    EEDetId cell(ix, iy, iz);
+    return cell;
+  } else {
     std::cout << "Null coordinates = " << ix << "," << iy << "," << iz << std::endl;
```
Comment (tvami) [inline CalibCalorimetry/CaloMiscalibTools/src/MiscalibReaderXMLEcalEndcap.cc:16]: `cout` should be replaced with the relevant MsgLogger function, maybe for another PR tho?

File: CalibCalorimetry/EcalCorrelatedNoiseAnalysisAlgos/src/TEcnaWrite.cc [hunk 2]
```diff
@@ -90,10 +90,10 @@ TEcnaWrite::TEcnaWrite(TEcnaObject* pObjectManager, const TString& SubDet) {
...
 TEcnaWrite::TEcnaWrite(const TString& SubDet,
-                       const TEcnaParPaths* pCnaParPaths,
-                       const TEcnaParCout* pCnaParCout,
-                       const TEcnaParEcal* pEcal,
-                       const TEcnaNumbering* pEcalNumbering) {
+                       TEcnaParPaths* pCnaParPaths,
+                       TEcnaParCout* pCnaParCout,
+                       TEcnaParEcal* pEcal,
+                       TEcnaNumbering* pEcalNumbering) {
   // std::cout << "[Info Management] CLASS: TEcnaWrite.         CREATE OBJECT: this = " << this << std::endl;
```
Comment (tvami) [inline CalibCalorimetry/EcalCorrelatedNoiseAnalysisAlgos/src/TEcnaWrite.cc:97]: Please remove commented out code

**Comment by tvami** [comment]
@thomreis if you prefer I'm fine approving this PR as it is, and you can have a dedicated PR for `cout`-s and commented out code in the future?

**Comment by thomreis** [comment]
Hi @tvami, yes the purpose of this PR was to address SA issues. Code cleaning can be done in a future PR.

**Comment by tvami** [comment]
@thomreis there is a few more things reported by the SA: 
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b3636e/25257/llvm-analysis/

**Comment by thomreis** [comment]
Yes, the PR does not fix all of the SA issues. Some of them are related to HCAL, the non-const global static variables are from ROOT macros and the non-const static variables I do not really know how to fix without touching the functionality of the code.

**Comment by tvami** [comment]
+1
 * tests pass, diffs in MsgLogger are understood
 * further cleanup agreed in another PR, see https://github.com/cms-sw/cmssw/pull/38235#issuecomment-1146083898 and https://github.com/cms-sw/cmssw/pull/38235#issuecomment-1146556904

**Comment by qliphy** [comment]
+1

**Pull request merged**